src/statistics.jl

```diff
@@ -365,32 +365,32 @@ function mean!(
     if length(w) != n
         throw(DimensionMismatch("The number of weights ($(length(w))) does not match the number of points for the median ($(n))."))
     end
-    copyto!(y, x0)
-    yold = allocate_result(M, median, y)
+    copyto!(q, x0)
+    yold = allocate_result(M, median, q)
     ytmp = copy(yold)
-    v = zero_tangent_vector(M, y)
+    v = zero_tangent_vector(M, q)
```

Comment:
`v` to `X` in this file. Probably useful to change `x` to `p` as elsewhere to avoid confusion.

Reply:
I left `x` when it was a vector of points on the manifold. but I changed `x0` to `p0`.